yarn-project/archiver/src/archiver/archiver.test.ts

```diff
@@ -114,36 +115,51 @@ describe('Archiver', () => {
       blockStartNumber = 1,
       numL1ToL2Messages = 3,
       maxEffects = 0,
+      previousArchive,
+    }: {
+      numBlocksPerCheckpoint?: number;
+      txsPerBlock?: number;
+      checkpointStartNumber?: CheckpointNumber;
+      blockStartNumber?: number;
+      numL1ToL2Messages?: number;
+      maxEffects?: number;
+      previousArchive?: AppendOnlyTreeSnapshot;
     } = {},
   ) => {
-    return await Promise.all(
-      Array.from({ length: numCheckpoints }, async (_, i) => {
-        const checkpointNumber = CheckpointNumber(i + checkpointStartNumber);
-        const startBlockNumber = BlockNumber(i * numBlocksPerCheckpoint + blockStartNumber);
-        const endBlockNumber = BlockNumber(startBlockNumber + numBlocksPerCheckpoint - 1);
-        return await mockCheckpointAndMessages(checkpointNumber, {
-          startBlockNumber,
-          numBlocks: numBlocksPerCheckpoint,
-          txsPerBlock,
-          numL1ToL2Messages,
-          timestamp: BigInt(now + Number(ETHEREUM_SLOT_DURATION) * (endBlockNumber + 1)),
-          makeBlockOptions: blockNumber => ({
-            // State reference can't be random. The nextAvailableLeafIndex of the note hash tree must be big enough to
-            // avoid error when computing the dataStartIndexForBlock in LogStore.
-            state: makeStateReference(0x100),
-            timestamp: BigInt(now + Number(ETHEREUM_SLOT_DURATION) * (blockNumber + 1)),
-            txOptions: {
-              numPublicCallsPerTx: blockNumber + 1,
-              numPublicLogsPerCall: 2,
-              maxEffects,
-            },
-            makeTxOptions: txIndex => ({
-              numPrivateLogs: blockNumber + txIndex,
-            }),
+    // Create checkpoints sequentially to chain archive roots properly.
+    // Each checkpoint's first block's lastArchive must equal the previous block's archive.
```

Comment:
Ah damn, sorry that the new check required so much additional work on the test suite